src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,132 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
     @Override
     public void add(T value) {
+        Node<T> t = tail;
+        Node<T> newNode = new Node<>(t, value, null);
+        tail = newNode;
+        if (t == null) {
+            head = newNode;
+        } else {
+            t.next = newNode;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index == size) {
+            add(value);
+        } else {
+            Node<T> var = getNode(index).prev;
+            Node<T> newNode = new Node<>(var, value, getNode(index));
+            getNode(index).prev = newNode;
+            if (var == null) {
+                head = newNode;
+            } else {
+                var.next = newNode;
+            }
+            size++;
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T value : list) {
+            add(value);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        return getNode(index).value;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        final T element = getNode(index).value;
+        Node<T> var = getNode(index);
+        var.value = value;
+        return element;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        final T element = getNode(index).value;
+        Node<T> prev = getNode(index).prev;
+        Node<T> next = getNode(index).next;
```

Comment:
```suggestion
		Node<T> target = getNode(index);
        final T element = target.value;
        Node<T> prev = target.prev;
        Node<T> next = target.next;
```
так по перформенсу буде краще так як ітерація нод може бути доволі затратною операцією